lib/Doctrine/DBAL/Driver/IBMDB2/DB2Connection.php

```diff
@@ -175,4 +175,18 @@ public function errorInfo()
             1 => $this->errorCode(),
         );
     }
+
+    /**
+     *
+     * Retrieves ibm_db2 native resource handle.
+     *
+     * Could be used if part of your application is not using DBAL.
+     *
+     * @return resource
+     */
+    public function getConn() //getWrappedResourceHandle
```

Comment:
Fixed the method name.
Here is the Zend XMLToolkit docs using directly the resource handler.
The fact is on IBMi system once you get a connection to the system it's not just a simple connection but a lot of system services are created and bound to it.
For example a temporary schema (QTEMP) which is avaiable only using that connection, so a SQL statement won't be able to access temporary tables created by another connection (the case you proposed).
Another example is about Library Lists on IBMi. If you use the connection parameter i5_naming=system, the system will look for a table not on a single schema but on a list of schemas, using the first match. 
If a program or a XMLTookit command is execute to change the library list it will affect which table is queried. This is also bound to the connection, so another connection resource wouldn't reflect the system changes.
I hope this could answer to your doubts. I know IBMi system kind of complex and unfortunately not so standard.